### ISSUE TITLE
Autotune Admin

### ISSUE THREAD

**Issue by ChakshuGautam** [BharatSahAIyak/autotune#22]
- [ ] Allow for updates on datasets
- [ ] Form flow (iterative)

## User Stories
1. User is shown a login page to login using their Google account. (The account has to be of 'samagragovernance.in' domain). The user is then shown a settings page where they are nudged to update the API keys for OpenAI and HuggingFace. The user is also shown a list of all the repos they have access to on HuggingFace. The user can select the repo they want to work on and the settings are saved. The settings tab allows the user to view and update the settings.
2. User gives a prompt and samples are 5 generated automatically.
3. User can view the 5 generated samples and select the ones they like and dislike.
4. The prompt is updated with those examples and the process is repeated until the user is satisfied with the 5 samples.
5. Once the user is satified, user provides the number of samples they want and the data is generated. The process is async and progress is shown to the user. The progress is tracked every 2 seconds.
6. The data is generated and shown to the user, they give a go ahead and the dataset is pushed to huggingface. The dataset tab allows the user to view all the datasets that they have deployed until now on huggingface.
7. A link is shared with the user so that they can view the data on huggingface.
8. There is a tab called train, which allows user to use the dataset created earlier to train a model by filling a form. The process is async and progress is shown to the user. The progress is tracked every 2 seconds.
9. Once trained the user is allowed to view the results of the model on the test set. The user can also view the results of the model on the validation set. The models tab allows the user to view all the models that they have deployed until now.
10. The user is then nudged to deploy the model to huggingface. Once confirmed, the user is asked to provide a name for the model and the model is pushed to huggingface.
11. A link is shared with the user so that they can view the model on huggingface and a curl is shared so that they can use the model for inference.
12. The history tab allows the user to view all the tasks that they have performed until now.

**Comment by karntrehan** [comment]
* What if we want to move away from Huggingface later?  - Make it configurable.

**Comment by KDwevedi** [comment]
@ChakshuGautam can we close this as outdated?

**Comment by ChakshuGautam** [comment]
Yes.
